### PR TITLE
fix: update the logic of secondary vni to support multiple vsi_per_subnet

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Changes to the map structure used for creating the secondary VNIs.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by toddgiguere — APPROVED** [review]

**Comment by Aashiq-J** [comment]
The secondary VNI will be re-created which causes the upgrade test to fail.
hence, skipping the upgrade test.

File: main.tf [hunk 1]
```diff
@@ -27,15 +27,35 @@ locals {
...
-        vsi_name    = "${var.prefix}-${substr(var.subnets[subnet].id, -4, 4)}-${format("%03d", count + 1)}"
-        subnet_id   = var.subnets[subnet].id
-        zone        = var.subnets[subnet].zone
-        subnet_name = var.subnets[subnet].name
+        name           = "${var.subnets[subnet].name}-${count}"
+        vsi_name       = "${var.prefix}-${substr(var.subnets[subnet].id, -4, 4)}-${format("%03d", count + 1)}"
+        subnet_id      = var.subnets[subnet].id
+        zone           = var.subnets[subnet].zone
+        subnet_name    = var.subnets[subnet].name
+        secondary_vnis = [for index, vni in ibm_is_virtual_network_interface.secondary_vni : vni.id if(vni.zone == var.subnets[subnet].zone) && (tonumber(substr(index, -1, -1)) == count)]
```
[the remark this answers is not in the document]
Comment (Aashiq-J) [reply main.tf:35]: Yes, I had asked Nupur from hpc team to test this branch and she has said she didn't face any issue with this branch.

File: main.tf [hunk 2]
```diff
@@ -238,11 +259,11 @@ resource "ibm_is_instance" "vsi" {
 
   # Additional Virtual Network Interface
   dynamic "network_attachments" {
-    for_each = { for key, value in ibm_is_virtual_network_interface.secondary_vni : key => value if key == each.value.zone && !var.use_legacy_network_interface }
+    for_each = { for index, id in each.value.secondary_vnis : index => id if !var.use_legacy_network_interface }
     content {
-      name = network_attachments.value.name
+      name = "${each.value.vsi_name}-secondary-vni-${network_attachments.key}"
```
[the remark this answers is not in the document]
Comment (Aashiq-J) [reply main.tf:264]: Yes, the value contains the id of the secondary VNI.

**Review by toddgiguere — APPROVED** [review]

**Pull request merged**